scripts/toolchain.sh

```diff
@@ -40,3 +40,13 @@ ____________EOF
 fi
 
 popd >/dev/null
+
+DEFAULT_CLANG_TIDY_VERSION=13
+export CLANG_TIDY="${CLANG_TIDY:-$(which "clang-tidy-$DEFAULT_CLANG_TIDY_VERSION" || which clang-tidy)}"
+
+if [ ! "$CLANG_TIDY" ]; then
+    printf '\033[33m[WARNING] Clang Tidy ('"$DEFAULT_CLANG_TIDY_VERSION"') has not been installed.\033[0m\n' >&2
+elif ! "$CLANG_TIDY" --dump-config | grep -i '^[[:space:]]*Checks[[:space:]]*:' > /dev/null; then
```

Comment:
FYI usually I just --version. I'm not sure if all of them have --dump-config.

Reply:
And be caution, check if apple clang has tidy because it's 13 would be different from the llvm's 13.

Reply:
> FYI usually I just --version. I'm not sure if all of them have --dump-config.

I tried, but there are no "clang-tidy" that I can grep from the `--version` output, so it is actually too weak.

```
> clang-tidy-13 --version
Ubuntu LLVM version 13.0.1
  
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: haswell
```

> check if apple clang has tidy because its 13 would be different from the llvm's 13. They all have their own versioning.

Apple clang has no tidy. You can install LLVM's clang-tidy from homebrew though.